docs(contributing): Add Documentation Style Guide

aimeeu opened the pull request:
This is the first of 4 PRs to add info on contributing to the Spinnaker documentation. Before the end of March, I plan to add pages on editing existing docs content, adding new docs content, and reviewing docs PRs,  coauthored with Brian Le.

**Note**: The TOC conflicts with a table definition in the content, so the style guide page is rendered with a huge gap near the top of the Language section. I haven't figured out why this is happening or what to do about it. I may rewrite, omitting all the tables in the content, which is probably a better idea since some tables only have a single row at the point.

This docs style guide is based on Kubernetes' [Documentation Style Guide](https://kubernetes.io/docs/contribute/style/style-guide/) and the [Google developer documentation style guide](https://developers.google.com/style). 

I would appreciate feedback on the content as well as missing topics that you'd like added, especially in the _Spinnaker.io word list_ section.

Partial: Issue #1707 
Fix: Issue #1730

aimeeu committed:
WIP docs(community): Add docs style guide

Signed-off-by: Aimee Ukasick <aimeeu.opensource@gmail.com>

aimeeu committed:
docs(community): add style guide

Added final paragraphs.

aimeeu committed:
docs(contributing): add docs style guide

Add new 'docs' directory under contributing.
Add to navigation.yml

aimeeu commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,292 @@
+| Open the `/.hal/config` file. | Open the /.hal/config file.|
+
+### Lists
+Group items in a list that are related to each other. Use a numbered list for items that need to appear in a specific order.
```
Better would be: Use a numbered list for instructions that need to be completed in a specific order.

317brian commented:
IIRC We've had this TOC issue w/ other pages too, which is why I'm so adamant about a better TOC in the revamped site.

317brian commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,292 @@
+
+## Language
+
+The Spinnaker documentation uses U.S. English spelling and grammar. However, use the international standard for punctuation inside quotes.
```
I don't think people would know what `the international standard for punctuation inside quotes` is. I don't.

dorbin replied:
I don't either, but it's illustrated below. Maybe if you just end the sentence above with a colon, you'd imply that _this thing here_ is the standard. 

Or add a link to the standard?

317brian commented:
The reference style link is the thing causing the build to fail: https://s.armory.io/eDux8qpe

It 404s

317brian commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,292 @@
+~~~~~~~~~
+
+### Line breaks
+U#se a single newline to separate block-level content like headings, lists, images, code blocks, paragraphs, and others.
```
Random # in `Use`

aimeeu committed:
Change ref-style-link declaration.

Change ref-style-link to RefStyleLinkDoc.
Minor wording change in Lists section.

aimeeu committed:
docs(contributing): docs style guide

Removed ref style link because job consistently failed, unable to find
the declaration at the bottom of the file.

aimeeu commented:
I don't know why the complier couldn't find the reference style link declaration at the bottom of the page . The URL was correct. Tried renaming the reference link declaration but same error. So I removed it and put the link in the body text.

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+
+### Avoid statements about the future
+
+If you need to write about an alpha feature, put the text in a note or under a heading that identifies it as alpha information.
```
```suggestion
If you need to write about an alpha feature, put the text in a note or under a heading that identifies it as alpha information, but don't include statements about when the feature will no longer be alpha, or what will change.
```

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+### Use sentence capitalization for headings
+
+~~~~~~~~~
+# Creating a custom webhook stage
```
I can't find it atm in the Google dev doc style guide, so maybe I'm off base, but...
I think we want to encourage imperative verbs in headings rather than gerunds, though we prefer gerunds in article titles.

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+Group items in a list that are related to each other. Use a numbered list for instructions that need to be completed in a specific order.
+
+ - End each item in a list with a period if one or more items in the list are complete sentences. For the sake of consistency, normally either all items or none should be complete sentences.
+ - Use the number one (1.) for ordered lists.
```
```suggestion
 - Use the number `1.` for each item in ordered lists.
```

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+ - Use the number one (1.) for ordered lists.
+ - Use (+), (* ), or (-) for unordered lists.
+ - Leave a blank line after each list.
+ - Indent nested lists with four spaces (for example, ⋅⋅⋅⋅).
```
Is that the correct md rule? I always just assumed it's the number of spaces required to line up exactly under the text above. in fact, for unordered lists I use three spaces (cause the * only takes up one space).

aimeeu replied:
I double checked online markdown resources - most state one tab or four spaces.  Brian has a point in that it depends on the Markdown converter.  The [Kramdown](https://kramdown.gettalong.org/quickref.html) docs don't state tabs or spaces.

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+ - Use (+), (* ), or (-) for unordered lists.
+ - Leave a blank line after each list.
+ - Indent nested lists with four spaces (for example, ⋅⋅⋅⋅).
+ - List items may consist of multiple paragraphs. Each subsequent paragraph in a list item must be indented by either four spaces or one tab.
```
Uh oh. Spaces v. tabs. 🚀 

But my same question applies. For list paragraphs, I indent with spaces to line up.

aimeeu replied:
So do I. For people new to markdown, though, having a concrete guideline is perhaps better.

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+|**Do** |**Do Not**|
+|:------|:---------|
+Set the value of `enabled` to True. | Set the value of `enabled` to "True".
+Set the value of `image` to nginx:1.8. | Set the value of `image` to `nginx:1.8`.
```
But code-font/backtics, right?

Set the value of `enabled` to `True`. 
Set the value of `image` to `nginx:1.8`.

aimeeu replied:
Excellent catch - updated

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+
+| **Do**                         | **Do Not**                                                         |
+|:------------------------------ |:------------------------------------------------------------------ |
+| Kubernetes                     | kubernetes                           |
```
```suggestion
| Kubernetes                     | kubernetes, k8s                       |
```

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+| Kubernetes                     | kubernetes                           |
+| Docker                         | docker                               |
+| SIG Docs                       | SIG Docs rather than SIG-DOCS or other variations.                 |
+| On-premises                    | On-premises or On-prem rather than On-premise or other variations. |
```
```suggestion
| On-premises, On-prem           |  On-premise or other variations. |
```

dorbin commented on community/contributing/docs/docs-style-guide.md:
```diff
@@ -0,0 +1,290 @@
+
+This page gives writing style guidelines for the Spinnaker documentation. Since these are guidelines, not rules, use your best judgment when creating content. Feel free to propose changes to this document in a pull request.
+
+If you have a style question that isn't answered in this guide, ask the SIG Docs team in the `#sig-documentation` channel.
```
This assumes the reader/contributor is already familiar with the Spinnaker Slack workspace. Is that a safe assumption? (It may well be; I'm just wondering.)

dorbin reviewed (approved):
@aimeeu , sorry it's taken me a few days to get to this! It looks great!

I'd like to merge this fairly quickly and iterate on it. In fact, I don't think any of my suggestion are important enough to hold you up. So I'm approving this but not merging it, and you can make the changes, save them for later, or disagree with them

dorbin committed:
Merge branch 'master' into docs-style-guide

317brian committed:
Merge branch 'master' into docs-style-guide

aimeeu committed:
Addressed Dave's feedback

317brian committed:
Merge branch 'master' into docs-style-guide

317brian reviewed (approved):
Thanks for putting this together and addressing our concerns. Great job @aimeeu !
